tasks/debian-install.yml

```diff
@@ -0,0 +1,224 @@
+
+---
+
+# Step 1: Validation of Variables
+- name: assert all vars are set
+  assert: { that: "{{ item }} != '' " }
+  with_items:
+  - postgresql_version
+  - postgresql_port
+  - postgresql_user
+  - postgresql_password
+  - postgresql_group
+  - postgresql_service_name
+  - postgresql_packages
+  - postgresql_data_directory
+  - postgresql_bin_directory
+  - postgresql_conf
+  - postgresql_dependencies
+  - postgresql_service_name
+
+# Step 2: Add PostgreSQL repository
+- name: Add PostgreSQL APT repository
+  apt_repository:
+    repo: "deb http://apt.postgresql.org/pub/repos/apt/ {{ ansible_distribution_release }}-pgdg main"
+    state: present
+  become: yes
+
+# Step 3: Update APT cache
+- name: Update APT cache
+  apt:
+    update_cache: yes
+  become: yes
+
+# Step 4: Install PostgreSQL dependencies
+- name: Install PostgreSQL dependencies
+  apt:
+    name: "{{ postgresql_dependencies }}"
+    state: present
+  become: yes
+
+# Step 5: Install PostgreSQL
+- name: Install PostgreSQL
+  apt:
+    name: "{{ postgresql_packages }}"
+    state: present
+  become: yes
+
+# Step 6: Initialize PostgreSQL Server
+- name: Make sure the postgres data directory exists
+  file:
+    path: "{{ postgresql_data_directory }}"
+    state: directory
+    owner: "{{ postgresql_user }}"
+    group: "{{ postgresql_group }}"
+    mode: 0700
+  become: yes
+
+- name: Check if PostgreSQL data directory is initialized
+  stat:
+    path: "{{ postgresql_data_directory }}/PG_VERSION"
+  register: pgdata_dir_initialized
+  become: yes
+  
+- name: Ensure directory /home/{{ postgresql_user }}/.ansible/tmp exists
```

Comment:
what is anti pattern?

Reply:
Name is placeholder for a task. Making the task name dynamic makes ansible hard to be idempotent. 

Including variables in task names can make it harder to read and maintain Ansible playbooks because it blurs the distinction between task names and task parameters. It can also lead to issues if the variable value changes unexpectedly or if you try to reference a variable that hasn't been defined yet.